panic: "id <= evictCount" (http2/hpack) when passing Fiber zero-copy values into gRPC metadata

### Feature Proposal Description

### Summary

Retaining a Fiber zero-copy value (`c.Get`, `c.Cookies`, `c.Query`, `c.Params`,
`c.Body`) past the handler — specifically by forwarding it into a long-lived gRPC /
HTTP/2 client's metadata — corrupts the HTTP/2 HPACK dynamic table and crashes the
process with:

```
panic: id (N) <= evictCount (M)
golang.org/x/net/http2/hpack.(*headerFieldTable).idToIndex
...
google.golang.org/grpc/internal/transport.(*loopyWriter).writeHeader
created by google.golang.org/grpc/internal/transport.NewHTTP2Client
```

This is **working as intended** for Fiber/fasthttp (zero allocation), and the
godoc already says "make copies or use Immutable". The gap is **discoverability**:
the panic surfaces in `x/net/http2/hpack` with nothing pointing back to Fiber, so
people repeatedly hit it and conclude "Fiber is broken" / switch frameworks
(see #2067, which was closed without a minimal repro).

### Proposal

Add a short, **searchable** "Common pitfall" callout to the Zero Allocation docs
that names the panic text and shows the two fixes. Searching the panic string
should land on this page.

### Why it recurs

`c.Get()` returns a string aliasing fasthttp's request buffer. gRPC's HPACK
encoder indexes that string into its connection-lifetime dynamic table. When
fasthttp reuses the buffer for the next request, the referenced bytes mutate, the
table's internal maps desync, and the next header encode panics in the
`loopyWriter` goroutine. The corruption is single-goroutine, so `-race` does not
catch it.

### Minimal reproduction

A self-contained `go run .` reproduction (Fiber handler → long-lived gRPC client,
no `.proto`/codegen needed) is attached/linked. Without a fix it panics in a few
seconds; `FIX=clone` (`strings.Clone`) and `FIX=immutable` (`Immutable: true`)
both run cleanly. Pinned to `fiber v2.52.12`, `grpc v1.79.3`, `x/net v0.56.0`.

### Related

golang/go#43051, golang/go#72940, golang/go#69316, grpc/grpc-go#3957,
valyala/fasthttp#1368, and the closed gofiber/fiber#2067.


### Alignment with Express API

.

### HTTP RFC Standards Compliance

.

### API Stability

.

### Feature Examples

```go
https://github.com/octaview/fiber-hpack-repro
```

### Checklist:

- [x] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/main/.github/CODE_OF_CONDUCT.md).
- [x] I have searched for existing issues that describe my proposal before opening this one.
- [x] I understand that a proposal that does not meet these guidelines may be closed without explanation.

You're absolutely right, and I'm not suggesting otherwise — Immutable and the per-method godoc ("make copies or use the Immutable setting") are clear, and it's on the developer to read them. I'm not proposing a code change or calling this a bug.

The narrow gap I ran into is discoverability from the symptom side. The docs are organized by cause ("Zero Allocation"), but when this actually bites, the developer sees a panic in a totally unrelated package with zero mention of Fiber:

panic: id (N) <= evictCount (M)
golang.org/x/net/http2/hpack.(*headerFieldTable).idToIndex
...
google.golang.org/grpc/internal/transport.NewHTTP2Client

There's nothing in that stack trace to make someone search the Fiber docs — so they keep landing in issues instead (#2067, plus golang/go#43051, grpc/grpc-go#3957, fasthttp#1368), and some just conclude "Fiber is broken" and switch frameworks.

So the proposal is just a one-paragraph callout in the existing Zero Allocation section that names the panic string, so a web search for it lands on the docs that already explain the fix. Nothing about the behavior or the responsibility changes — it's purely a searchable signpost back to what's already documented.

I put together a self-contained go run . reproduction (panics in seconds; strings.Clone and Immutable: true both fix it) if a concrete repro is useful: <link>. Totally fine if you'd still rather not add it — just thought the recurring reports made it worth a signpost.

--- a/docs/extra/faq.md
+++ b/docs/extra/faq.md
@@ -198,3 +198,24 @@ For details on how to:
 * Convert `fiber.Ctx` to `http.Request`
 
 See the dedicated documentation: [Adaptor Documentation](../middleware/adaptor.md).
+
+## Troubleshooting common errors
+
+Crashes that surface in unrelated packages, with stack traces that never mention Fiber, usually trace back to a context value kept past the handler. The cases below name the symptom so a search for the panic text lands here.
+
+### Panic `id (N) <= evictCount (M)` (hpack / HTTP/2 / gRPC) {#panic-id-evictcount-hpack-grpc}
+
+This is not a bug in Fiber, gRPC, or `golang.org/x/net/http2`. It means a Fiber zero-copy context value (a header from `c.Get`, or `c.Query`, `c.Params`, `c.Cookies`, `c.Body`) was kept past the handler. Fiber reuses the request buffer on the next request, so the retained bytes change underneath whatever stored them. See [Zero Allocation](../intro.md#zero-allocation) for the underlying behavior.
+
+A common trigger is forwarding a header into a long-lived gRPC / HTTP/2 client's `metadata`. The value is held in the connection's HPACK dynamic table and later mutates, crashing the process inside the HPACK encoder, in a stack trace that never mentions Fiber:
+
+```text
+panic: id (N) <= evictCount (M)
+    golang.org/x/net/http2/hpack.(*headerFieldTable).idToIndex
+    ...
+    google.golang.org/grpc/internal/transport.(*loopyWriter).writeHeader
+```
+
+The crash is decoupled from your handler in both time and call stack, so ordinary tests and even the `-race` detector usually miss it.
+
+The fix is to detach the value before you keep it: copy it with `utils.CopyString` (or `strings.Clone`), or enable [`Immutable`](../api/fiber.md#immutable). A full reproduction is in [gofiber/fiber#4464](https://github.com/gofiber/fiber/issues/4464).

